pkg/gateway/model_build_servicenetwork.go

```diff
@@ -25,25 +27,30 @@ type ServiceNetworkModelBuilder interface {
 }
 
 type serviceNetworkModelBuilder struct {
+	client      client.Client
 	defaultTags map[string]string
 }
 
-func NewServiceNetworkModelBuilder() *serviceNetworkModelBuilder {
-	return &serviceNetworkModelBuilder{}
+func NewServiceNetworkModelBuilder(client client.Client) *serviceNetworkModelBuilder {
+	return &serviceNetworkModelBuilder{client: client}
 }
 func (b *serviceNetworkModelBuilder) Build(ctx context.Context, gw *gateway_api.Gateway) (core.Stack, *latticemodel.ServiceNetwork, error) {
 	stack := core.NewDefaultStack(core.StackID(k8s.NamespacedName(gw)))
-
+	vpcAssociationPolicy, err := GetAttachedPolicy(ctx, b.client, k8s.NamespacedName(gw), &v1alpha1.VpcAssociationPolicy{})
```

Comment:
what happens if the policy is not found?

Reply:
If policy not found, GetAttachedPolicy() retrun nil, nil, it will go ahead to execute the rest of logic. Unit test covered that

 https://github.com/aws/aws-application-networking-k8s/blob/520448301016d569a458394b05ebc9e94e4f2ffb/pkg/gateway/utils_test.go#L205